gen/templates/resource.go.tmpl

```diff
@@ -638,6 +638,15 @@ func set{{ .ResourceClassName }}LegacyAttributes(ctx context.Context, diags *dia
 			{{- end}}
 		{{- end }}
 	{{- end }}
+	{{- $changed := false}}
+		{{- range .LegacyAttributes}}
+		{{- if ne .ReplacedBy.AttributeName "" }}
```

Comment:
I'm not sure if this is relevant but should we reset changed to false every time before the start of the iteration of the loop?
```
{{- range .LegacyAttributes}}
{{- $changed := false}}
```

Reply:
yes it is once to determine that there are legacy attributes with a name change